### PR TITLE
Display Cliqz module list

### PR DESCRIPTION
**Pull request by chrmod**
This PR attempts to list Cliqz modules in the Elm app.

### CODE REVIEW

**Comment by chrmod** [comment]
Elm type generator does not handle type `Array<string>`, it throws this error:
```

    ERROR in ./src/elm/Broxy.elm
    Module build failed: Error: Compiler process exited with error Compilation failed
    -- NAMING ERROR ---------------------------------- ./src/elm/TsElmInterfaces.elm
    
    Cannot find variable `decodeJEncode`
    
    46|         |: JDecode.field "modules" decodeJEncode.Value_is_not_implemented
                                           ^^^^^^^^^^^^^
    
    
    Detected errors in 1 module.
    
        at ChildProcess.<anonymous> (/home/chrmod/Projects/cliqz-oss/broxy/node_modules/node-elm-compiler/index.js:141:27)
        at emitTwo (events.js:125:13)
        at ChildProcess.emit (events.js:213:7)
        at maybeClose (internal/child_process.js:927:16)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
     @ ./src/static/renderer.ts 6:14-37
```

File: src/elm/Broxy.elm [hunk 1]
```diff
@@ -38,11 +42,11 @@ type Msg
 
 init : ( Model, Cmd Msg )
 init =
-    ( { proxy = Nothing, error = Nothing }, requestISocksProxy () )
+    ( { proxy = Nothing, error = Nothing, modules = Nothing }, requestISocksProxy () )
```
Comment (chrmod) [inline src/elm/Broxy.elm]: have no idea on how to subscribe for two ports in init

File: src/elm/Broxy.elm [hunk 2]
```diff
@@ -29,6 +32,7 @@ main =
 type alias Model =
     { proxy : Maybe ISocksProxy
     , error : Maybe String
```
Comment (chrmod) [inline src/elm/Broxy.elm:29]: as there can be multiple subscriptions, singular error value is not sufficient. 
I'm thinking on how encapsulate ISocksProxy and its error into type

Comment (remusao) [reply src/elm/Broxy.elm:29]: One way would be to a separate sum type for errors:
```elm
type Error =
    ProxyStartupError String
  | AdblockerStartupError String
  ...
```

Then the `Model` could become:
```elm
type alias Model =
    { proxy : Maybe ISocksProxy
    , error : Maybe String
    }
```

Although, now that I think about it... we should probably decide what should be in the `Model`. Does it even make sense to have an `error` field? It all depends on what we want to display in the UI I guess?

File: src/elm/Broxy.elm (hunk 1 above)
Comment (remusao) [reply src/elm/Broxy.elm]: @chrmod http://package.elm-lang.org/packages/elm-lang/core/5.1.1/Platform-Cmd

Commands can be batched, so using `Cmd.batch` we can provide a list of commands. That should do the trick.

**Comment by remusao** [comment]
@chrmod Regarding the type generation, that looks like a bug. I'll fix it.

**Comment by remusao** [comment]
@chrmod Interestingly, it seems that the typescript parser does not handle `Array<string>` and `string[]` the same way, although they should be equivalent. Also, it seems the typescript linter discourages the use of `Array<string>` (for some reason that I don't know). So as a work-around, I would suggest we use `string[]` instead, which is supported by the type generation. I will come up with a fix anyway.

File: src/elm/Broxy.elm (hunk 2 above)
Comment (chrmod) [reply src/elm/Broxy.elm:29]: have no opinion on that yet. 

but atm the UI is reacting to messages from the background so technically there is now way to get an error, right?

Comment (remusao) [reply src/elm/Broxy.elm:29]: No exception should come from the Elm side, so we would only reflect exceptions from Cliqz in the view.

**Review by remusao — APPROVED** [review]

**Pull request merged**